### PR TITLE
Use x86 machines for ios builds.

### PR DESCRIPTION
**Pull request by godofredoc**
This is to workaround rosetta not being installed in dart-internal machines and to unblock beta releases.

Bug: https://github.com/flutter/flutter/issues/136219

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by zanderso** [comment]
What is the plan for getting Rosetta onto the release builders?

**Review by XilaiZhang — APPROVED** [review]

File: ci/builders/mac_ios_engine.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
             "drone_dimensions": [
                 "device_type=none",
                 "mac_model=Macmini8,1",
-                "os=Mac-12"
+                "os=Mac-12",
```
Comment (zanderso) [inline ci/builders/mac_ios_engine.json:7]: Some of these have `"mac_model=Macmini8,1"` and some don't. Is that important?

**Review by zanderso — APPROVED** [review]
LGTM to help get the release out, but I would like to know what the plan is for making the all of the CI environments consistent with each other.

File: ci/builders/mac_ios_engine.json (hunk 1 above)
Comment (godofredoc) [reply ci/builders/mac_ios_engine.json:7]: Good point mac_model=Macmini8,1 will resolve to x86. Removing them for consistency.

**Comment by godofredoc** [comment]
> Macmini8



> LGTM to help get the release out, but I would like to know what the plan is for making the all of the CI environments consistent with each other.

It is unlikely we'll be able to enable rosetta on dart-internal machines. Opened https://github.com/flutter/engine/pull/46697 to run some validations and it seems like the only blocker is gen_snapshot expected to be x64, I'll talk to @cbracken about potential solutions.

**Comment by zanderso** [comment]
> It is unlikely we'll be able to enable rosetta on dart-internal machines.

Could you explain why this is? I think it is very important that the release builder environment and the prod/staging/try environments are consistent with each other, or we are going to have some very confusing issues come up only when we are doing release builds.

**Comment by godofredoc** [comment]
> > It is unlikely we'll be able to enable rosetta on dart-internal machines.
> 
> Could you explain why this is? I think it is very important that the release builder environment and the prod/staging/try environments are consistent with each other, or we are going to have some very confusing issues come up only when we are doing release builds.

Internal macs are provisioned with additional restrictions to comply with higher security requirements and the provisioning process is standardized for all the teams.

**Comment by zanderso** [comment]
> Internal macs are provisioned with additional restrictions to comply with higher security requirements and the provisioning process is standardized for all the teams.

Could you please clarify what is the specific security requirement that makes it impossible to have Rosetta installed on these machines?

**Comment by cbracken** [comment]
Just wanted to follow up on @zanderso's question as to what security concerns we have with Rosetta. Is there some documentation covering the rationale?

**Pull request merged**